src/sage/matroids/oriented_matroids/signed_subset_element.py

```diff
@@ -0,0 +1,850 @@
+                        self._n.add(label)
+                    elif j == 1 or j == '+':
+                        self._p.add(label)
+                    elif j == 0 or j == '' or j == '0':
+                        self._z.add(label)
+                    else:
+                        raise ValueError("must be tuple of -1, 0, 1")
+
+            # If we have a tuple of tuples
+            else:
+                self._p = set(data[0])
+                self._n = set(data[1])
+                if len(data) > 2:
+                    self._z = set(data[2])
+                elif groundset is not None:
+                    self._z = set(groundset).difference(
+                        self._p).difference(self._n)
+        # If we have a dictionary, use the keys to figure it out
+        elif isinstance(data, dict):
+            if 'p' in data:
+                self._p = data['p']
+            if 'positives' in data:
+                self._p = data['positives']
+            if 'n' in data:
+                self._n = data['n']
+            if 'negatives' in data:
+                self._n = data['negatives']
+            if 'z' in data:
+                self._z = data['z']
+            if 'zeros' in data:
+                self._z = data['zeros']
+        else:
+            raise ValueError(
+                "either positives and negatives are set or data is a tuple, OrientedMatroidELement or a dict")
+
+        # Type fix
+        self._p = set(self._p)
+        self._n = set(self._n)
+        self._z = set(self._z)
+
+        # Setup the groundset if it's not set yet
+        if groundset is None:
+            self._g = list(self._p.union(self._n).union(self._z))
+        else:
+            if not self.support().union(self.zeros()).issubset(groundset):
+                raise ValueError("elements must appear in groundset")
+
+            # Update the zeros with everything in the groundset
+            if self._z is None:
+                self._z = set(groundset).difference(self.support())
+
+            # groundset should be everything
+            if not set(groundset).issubset(self.support().union(self.zeros())):
+                raise ValueError(
+                    "every element must be either positive, negative or zero")
+            self._g = groundset
+
+        self._g = list(self._g)
+
+    def __call__(self, var):
+        """
+        Return the sign of an element in the groundset.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E = M.elements()[0]
+            sage: E(1)
+            0
+            sage: E(0)
+            1
+            sage: E(2)
+            -1
+        """
+        if var in self.positives():
+            return 1
+        if var in self.negatives():
+            return -1
+        if var in self.zeros():
```

Comment:
```suggestion
        if var in self._p:
            return 1
        if var in self._n:
            return -1
        if var in self._z:
```
This will be faster, and I feel this is a class where speed is important. Plus I think this is more natural with these attributes being part of the class.